crates/anemo-benchmark/src/main.rs

```diff
@@ -0,0 +1,230 @@
+    loop {
+        let result = client
+            .send_bytes(anemo::Request::new(bytes.clone()))
+            .await
+            .map(|_| peer_id);
+        notify.send((peer_id, result.err())).unwrap();
+    }
+}
+
+async fn download_from_peer(
+    peer: anemo::Peer,
+    size: u32,
+    notify: UnboundedSender<(anemo::PeerId, Option<anemo::rpc::Status>)>,
+) {
+    let peer_id = peer.peer_id();
+    let mut client = BenchmarkClient::new(peer);
+    loop {
+        let result = client.request_bytes(anemo::Request::new(size)).await;
+        notify.send((peer_id, result.err())).unwrap();
+    }
+}
+
+#[tokio::main]
+#[allow(clippy::disallowed_methods)] // unbounded_channel is ok for benchmark reporting
+async fn main() {
+    let args: Args = Args::parse();
+    let _guard = telemetry_subscribers::TelemetryConfig::new()
+        .with_env()
+        .init();
+
+    let mut config = anemo::Config::default();
+    let mut quic_config = anemo::QuicConfig::default();
+    quic_config.socket_send_buffer_size = args.socket_send_buffer_size;
+    quic_config.socket_receive_buffer_size = args.socket_receive_buffer_size;
+    config.quic = Some(quic_config);
+
+    let (_network, peers) = start_server(config, args.port, args.addrs.clone()).await;
+
+    let rng = rand::thread_rng();
+    let send_bytes: Vec<_> = rng
+        .sample_iter(rand::distributions::Standard)
+        .take(args.size_up as usize)
+        .collect();
+
+    let mut tasks = tokio::task::JoinSet::new();
+    let (upload_notify_tx, mut upload_notify) = tokio::sync::mpsc::unbounded_channel();
+    let (download_notify_tx, mut download_notify) = tokio::sync::mpsc::unbounded_channel();
+
+    for _ in 0..args.requests_up {
+        for peer in peers.iter().cloned() {
+            tasks.spawn(upload_to_peer(
+                peer,
+                send_bytes.clone(),
+                upload_notify_tx.clone(),
+            ));
+        }
+    }
+    for _ in 0..args.requests_down {
+        for peer in peers.iter().cloned() {
+            tasks.spawn(download_from_peer(
+                peer,
+                args.size_up,
+                download_notify_tx.clone(),
+            ));
+        }
+    }
+
+    let mut interval = tokio::time::interval(Duration::from_secs(args.tick_secs));
+    let mut upload_requests: usize = 0;
+    let mut upload_errors: usize = 0;
+    let mut upload_bytes: usize = 0;
+    let mut download_requests: usize = 0;
+    let mut download_errors: usize = 0;
+    let mut download_bytes: usize = 0;
+    let mut instant = interval.tick().await;
+
+    loop {
+        tokio::select! {
+            Some((peer_id, error)) = upload_notify.recv() => {
+                upload_requests += 1;
+                if let Some(error) = error {
+                    upload_errors += 1;
+                    println!("upload error on peer {peer_id}: {:?}", error);
+                } else {
+                    upload_bytes += args.size_up as usize;
```

Comment:
optional: it may be more robust to use just send back the response and add its actual size here.

Reply:
@mwtian @aschran I believe that Quinn status also offer the transferred bytes (up/down) over the wire. It might worth using or comparing against that as well which will contain the overall exchanged data.